core/trino-main/src/main/java/io/trino/operator/InterpretedHashGenerator.java

```diff
@@ -74,7 +96,33 @@ public String toString()
     {
         return toStringHelper(this)
                 .add("hashChannelTypes", hashChannelTypes)
-                .add("hashChannels", hashChannels)
+                .add("hashChannels", hashChannels == null ? "<identity>" : Arrays.toString(hashChannels))
                 .toString();
     }
+
+    public static InterpretedHashGenerator createPositionalWithTypes(List<Type> hashChannelTypes, BlockTypeOperators blockTypeOperators)
```

Comment:
move this method above constructors (used methods should be below users)

Reply:
Done